Content.Server/_CD/Spawners/ArrivalsSpawnPointSystem.cs

```diff
@@ -0,0 +1,63 @@
+using System.Linq;
+using Content.Shared.GameTicking;
+using Content.Shared.Roles;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server._CD.Spawners;
+
+public sealed class ArrivalsSpawnPointSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedMapSystem _mapSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<PlayerSpawnCompleteEvent>(OnPlayerSpawn);
+    }
+
+    private void OnPlayerSpawn(PlayerSpawnCompleteEvent args)
+    {
+        // Ensure they have a job, so that we won't end up making mobs spawn on arrivals.
+        if (args.JobId == null)
+            return;
+
+        // Get job, skip everything else if it ignores arrivals
+        var job = _prototypeManager.Index<JobPrototype>(args.JobId);
+        if (job.IgnoreArrivals)
+            return;
```

Comment:
```suggestion
        if (!_prototypeManager.TryIndex<JobPrototype>(args.JobId, out var job) ||
            job.IgnoreArrivals)
            return;
```
job lookup should be infallible, but just to be safe

Reply:
Made it two if statements, because it needs to be an 'and', and putting it in one if statement could allow it to be null.